keps/sig-api-machinery/00xx-admission-webhooks-to-ga.md

```diff
@@ -0,0 +1,625 @@
+
+The propose change is to add an ObjectSelector to the webhook API both in v1 and v1beta1:
+
+```golang
+type Webhook struct {
+    ...
+     // ObjectSelector decides whether to run the webhook on an object based
+     // on whether the object.metadata.labels matches the selector. A namespace object must
+     // match both namespaceSelector (if present) and objectSelector (if present).
+     // For sub-resources, if the labels on the newObject is effective (e.g. pods/status)
+     // they will be used, otherwise the labels on the parent object will be used. If the
+     // sub-resource does not have a parent object, the call may fail or skipped based on
+     // the failure policy.
+     //
+     // Default to the empty LabelSelector, which matches everything.
+     // +optional
+     ObjectSelector *metav1.LabelSelector `json:"objectSelector,omitempty" protobuf:"bytes,7,opt,name=objectSelector"`
+}
+```
+
+### Scope
+
+Current webhook Rules applies to objects of all scopes. That means a Rule can uses wildcards to target both namespaced and cluster scoped objects. The proposal is to add a scope field to Admission Webhook configuration to limit webhook target on namespaced object or cluster scoped objects. This enables webhook developers to target all namespace objects or all cluster-scoped objects. The field will be added to both v1 and v1beta1. The field is optional and empty value means no scope restriction.
+
+```golang
+type ScopeType string
+
+const (
+     // ClusterScope means that scope is limited to cluster objects.
+     ClusterScope ScopeType = "Cluster"
+     // NamespacedScope means that scope is limited to namespaced objects.
+     NamespacedScope ScopeType = "Namespaced"
+)
+
+type Rule struct {
+    ...
+
+     // Scope specifies the scope of this rule. If unespecified, the scope is
+     // not limited.
+     //
+     // +optional
+     Scope ScopeType `json:"scope,omitempty" protobuf:"bytes,3,opt,name=scope"`
+}
+```
+
+### http for localhost
+
+Admission Webhook requires https for all connections. This is not necessary when the webhook in running on the same machine as apiserver (like a side-car or static pod). By relaxing this restriction, it would be possible to run side-cars or static pods on the same machine without cert management. The API change for this feature is documentation and relaxing validation.
+
+Note that there must be caution around any validation relaxing in regard to rolling-back the cluster. It must be documented to the end user not to use the feature until they are not going to roll-back the cluster to a version that does not have this feature. If they do, they need to delete or update these webhook configuration as the old cluster does not allow running insecure webhooks as sidecar. To delete these configurations in an old API server, one must first fix the http validation problem otherwise finalizers would not allow deletion the object.
```

Comment:
more frequent line breaks

I would take a fix to validation in a service stream for validation that is ratcheting.